link/uprobe_test.go

```diff
@@ -293,3 +326,62 @@ func TestUprobeProgramCall(t *testing.T) {
 	// Assert that this time the value has not been updated.
 	assertMapValue(t, m, 0, 0)
 }
+
+func TestUprobeProgramCallSharedlib(t *testing.T) {
+	m, p := newUpdaterMapProg(t, ebpf.Kprobe)
+
+	// External symbol.
+	sym := "open"
+
+	// Load the '/bin/bash' executable.
+	ex, err := OpenExecutable("/bin/bash")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Obtain external symbol address.
+	cmd := fmt.Sprintf("/usr/bin/gdb /bin/bash --batch -ex 'printf \"%%p\", %s'", sym)
+	out, err := exec.Command("/bin/sh", "-c", cmd).Output()
```

Comment:
You should be able to invoke gdb directly, without going via `/bin/sh`.

Reply:
I tried several times without success. Is it possible that gdb in batch mode does some stuff with shell redirections / pipes / etc.?

Reply:
thanks 🙈